app/controllers/application_controller.rb

```diff
@@ -75,7 +74,18 @@ def set_current_user
     if current_user.present?
       current_user.update_last_access_at!
     end
-    yield
+    begin
+      yield
+    rescue RestClient::Exception => e
+      if e.http_code === 401
```

Comment:
I like this a _lot_ better.  Just to be on the safe side, we might want to say:
```suggestion
      if [401, 403].include?(e.http_code)
```

I'm _fairly_ certain that orchestration replies `401` in this case, but it also might differ by endpoint/service.  I've definitely never seen `5xx`, there have been [incidences](https://broad-institute.sentry.io/issues/3910539286/events/0024a467f66c4822a66db85abb765762/?project=1424198) of `403` that could potentially be related.  I'm a little iffy on this but it seems relatively safe to do so.

Reply:
Yeah, when the user needs to accept updated Terra ToS, I see 401 is the only HTTP status code used for the response from the Orchestration API.  That derives from [this line in Sam](https://github.com/broadinstitute/sam/blob/f8cb6929291d798f27185977f593b270a0a8cca1/src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardSamUserDirectives.scala#L118); and 401 is mapped to that "Unauthorized" code in [this line from the akka-http Scala library](https://github.com/akka/akka-http/blob/main/akka-http-core/src/main/scala/akka/http/scaladsl/model/StatusCode.scala#L132) (and, of course, the [HTTP RFC](https://www.rfc-editor.org/rfc/rfc9110.html#name-401-unauthorized)).